pkg/auth/providers/azure/config.go

```diff
@@ -0,0 +1,40 @@
+package azure
+
+import (
+	"strings"
+
+	"github.com/rancher/norman/types/slice"
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+)
+
+const (
+	globalAzureADGraphEndpoint = "https://graph.windows.net/"
+	globalMSGraphEndpoint      = "https://graph.microsoft.com"
+	chinaAzureADGraphEndpoint  = "https://graph.chinacloudapi.cn/"
+	chinaMSGraphEndpoint       = "https://microsoftgraph.chinacloudapi.cn"
+)
+
+func authProviderEnabled(config *v32.AzureADConfig) bool {
+	return config.Enabled && config.GraphEndpoint != ""
+}
+
+func graphEndpointDeprecated(config *v32.AzureADConfig) bool {
```

Comment:
Looks like you only use a single string field of config:
```suggestion
func graphEndpointDeprecated(endpoint string) bool {
```

Reply:
The purpose of changing the parameter type is not to make more sense with the function name. It's to only pass in the fields necessary, and avoid passing pointers, when it is convenient to do so. If it is changed to what I suggested then the current name works, and a comment can clarify that it is not the only way to determine this but the way we use.

Reply:
Adjusted.